L96-python-setuptools-plugin.md

```diff
@@ -0,0 +1,107 @@
+
+In order to prevent unintended behavior, the first thing the functionality will do is check the
+configuration file to see if the section has been configured. If the file does not exist or the section
+is not configured, no additional steps will be taken. `setuptools` will be provided with an empty
+list to indicate that there are no additional files to include in the distribution.
+
+## Rationale
+
+A downside of this proposal is that it will introduce yet another way to get the Python modules that
+correspond to the proto files.
+
+The primary way generate the Python files is using the
+[`grpc_tools.protoc` command line interface][protoc_cli]. The advantage of this mechanism of this
+method is also its disadvantage. It is a completely standalone way to generate the source code for
+the python modules. While projects are allowed to decide how to work this generated files in the way
+that best suits the project, there is no integration point. This means the project **must** decide
+how and when this command is executed, likely needing to write a script to execute this CLI.
+
+For projects that use `setuptools` to publish a package, `grpc-tools` includes a
+[custom command][grpc_command]. However, it also requires a specific call like the `grpc_tools.protoc`
+CLI and is not compatible with installing from a source distribution for projects that declare
+`setuptools` as build system that should be used. Additionally, one of the top result for "grpc
+setuptools command" is [this post][setuptools_gen_question], which suggests a custom command that
+doesn't rely on the one provided by `grpc-tools`. This suggest that there is a desire for functionality
+that doesn't require a specific all and that the existing functionality is under-documented.
+
+gRPC also provides a way to get the Python [classes at runtime][runtime_classes]
+([Proposal][runtime_proposal]). However, it is still [marked as experimental][runtime_api] almost 2
+years later and requires `grpc-tools` to be available at runtime. Similar to the current `setuptools`
+integration, it is not mentioned in the basic tutorial documentation.
+
+## Implementation
+
+- [ ] Proof of Concept with hard coded values.
+- [ ] Configuration file parsing & mapping to CLI arguments.
+
+Could be done later:
+
+- [ ] Migrate existing `grpc*` packages to use this functionality.
+
+I am planning work on it. I will be able to spend a few hours of each work week on this effort.
+
+## Open issues (if applicable)
+
+- Looking at the implementation of the [custom command][grpc_command], it looks as though only a
+  sub-set of the CLI arguments supported by `grpc_tools.protoc` need to be supported. I believe this
+  will become more clear after starting on the implementation. However, there could always be an
+  "additional arguments" type parameter to cover this if a project has a specific need. This would
+  also allow for CLI arguments to be passed to any gRPC plugins that might be used by a project.
```

Comment:
Good point. We don't need all of the arguments. Can we discuss what to support in one of the above comment?

Reply:
I added a TODO line to the new "Configuration" section. That would probably be the best place to start that conversation.